Clarify Start and End operators when boundaries are null (#104)
HL7/cql: input/pages/09-b-cqlreference.md

For the interval-point overload, the operator returns <span class="kw">true</span> if the given interval starts after the given point.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

For the interval-point overload, the operator returns <span class="kw">true</span> if the given interval ends before the given point.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

If the high boundary of the interval is open and non-null, this operator returns the <span class="kw">predecessor</span> of the high value of the interval. If the high boundary of the interval is open and <span class="kw">null</span>, this operator returns an _uncertainty_ from the <span class="kw">start</span> of the interval to the <span class="kw">maximum</span> value for the point type of the interval (inclusive).

If the high boundary is <span class="kw">null</span> and the interval point type is unknown, a choice of types, or <span class="id">Any</span>, then the result cannot be determined and this operator returns <span class="kw">null</span>.

If the argument is <span class="kw">null</span>, the result is <span class="kw">null</span>.

The following examples illustrate the behavior of the <span class="id">End</span> operator:

```cql
define "EndOfClosedInterval": end of Interval[1, 5] // 5
define "EndOfOpenHighInterval": end of Interval[1, 5) // 4
define "EndOfClosedNullHighInterval": end of Interval[1, null] // 2147483647

> define "EndOfOpendNullHighInterval": end of Interval[1, null) // uncertainty[1, 2147483647]

Reviewer 2: ```suggestion
define "EndOfOpenNullHighInterval": end of Interval[1, null) // uncertainty[1, 2147483647]
define "EndOfOpenNullHighOpenLowInterval": end of Interval(1, null) // uncertainty[2, 2147483647]
```

The <span class="id">End</span> operator returns the ending point of an interval.

If the high boundary of the interval is closed and non-null, this operator returns the high value of the interval. If the high boundary of the interval is closed and <span class="kw">null</span>, this operator returns the <span class="kw">maximum</span> value for the point type of the interval.

> If the high boundary of the interval is open and non-null, this operator returns the <span class="kw">predecessor</span> of the high value of the interval. If the high boundary of the interval is open and <span class="kw">null</span>, this operator returns an _uncertainty_ from the low bounday of the interval (using <span class="id">[Start](#start)</span> operator semantics to determine the low boundary) to the <span class="kw">maximum</span> value for the point type of the interval.

Reviewer 2: ```suggestion
If the high boundary of the interval is open and non-null, this operator returns the <span class="kw">predecessor</span> of the high value of the interval. If the high boundary of the interval is open and <span class="kw">null</span>, this operator returns an _uncertainty_ from the low boundary of the interval (using <span class="id">[Start](#start)</span> operator semantics to determine the low boundary) to the <span class="kw">maximum</span> value for the point type of the interval.
```

The <span class="kw">ends</span> operator returns <span class="kw">true</span> if the first interval ends the second. More precisely, if the starting point of the first interval is greater than or equal to the starting point of the second, and the ending point of the first interval is equal to the ending point of the second.

> This operator uses the semantics described in the <span class="kw">[start](#start)</span> and <span class="kw">[end](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="kw">[start](#start)</span> and <span class="kw">[end](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

For the interval-interval overload, if either argument is <span class="kw">null</span>, the result is <span class="kw">null</span>.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

For the interval-interval overload, if either argument is <span class="kw">null</span>, the result is <span class="kw">null</span>.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

The <span class="kw">intersect</span> operator for intervals returns the intersection of two intervals. More precisely, the operator returns the interval that defines the overlapping portion of both arguments. If the arguments do not overlap, this operator returns <span class="kw">null</span>.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

The <span class="kw">meets before</span> operator returns <span class="kw">true</span> if the first interval ends immediately before the second interval starts, while the <span class="kw">meets after</span> operator returns <span class="kw">true</span> if the first interval starts immediately after the second interval ends.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

The operator <span class="kw">overlaps before</span> returns <span class="kw">true</span> if the first interval overlaps the second and starts before it, while the <span class="kw">overlaps after</span> operator returns <span class="kw">true</span> if the first interval overlaps the second and ends after it.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

For the interval-point overload, this operator returns true if the interval contains (i.e. includes) the point, and the interval is not a unit interval containing only the point.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

For the point-interval overload, this operator returns true if the point is in (i.e. included in) the interval, and the interval is not a unit interval containing only the point.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```

If the low boundary of the interval is open and non-null, this operator returns the <span class="kw">successor</span> of the low value of the interval. If the low boundary of the interval is open and <span class="kw">null</span>, this operator returns an _uncertainty_ from the <span class="kw">minimum</span> value for the point type of the interval to the <span class="kw">end</span> of the interval (inclusive).

If the low boundary is <span class="kw">null</span> and the interval point type is unknown, a choice of types, or <span class="id">Any</span>, then the result cannot be determined and this operator returns <span class="kw">null</span>.

If the argument is <span class="kw">null</span>, the result is <span class="kw">null</span>.

The following examples illustrate the behavior of the <span class="id">Start</span> operator:

```cql
define "StartOfClosedInterval": start of Interval[1, 5] // 1
define "StartOfOpenLowInterval": start of Interval(1, 5] // 2
define "StartOfClosedNullLowInterval": start of Interval[null, 5] // -2147483648

> define "StartOfOpendNullLowInterval": start of Interval(null, 5] // uncertainty[-2147483648, 5]

Reviewer 2: ```suggestion
define "StartOfOpenNullLowInterval": start of Interval(null, 5] // uncertainty[-2147483648, 5]
define "StartOfOpenNullLowOpenHighInterval": start of Interval(null, 5) // uncertainty[-2147483648, 4]
```

The <span class="kw">starts</span> operator returns <span class="kw">true</span> if the first interval starts the second. More precisely, if the starting point of the first is equal to the starting point of the second interval and the ending point of the first interval is less than or equal to the ending point of the second interval.

> This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries have `null` values.

Reviewer: ```suggestion
This operator uses the semantics described in the <span class="id">[Start](#start)</span> and <span class="id">[End](#end)</span> operators to determine interval boundaries, particularly when those boundaries are `null`.
```
